### PR TITLE
feat(plasma-b2c): Handling property size for TextArea

### PR DESCRIPTION
**Pull request by Yakutoc**
Нужно добавить исправленную версию размеров и все что с этим связанно.

Было: Изменение свойства **size** никак не учитывалось в компоненте.

Стало:       

________________

:baby_chick: Download canary assets:

[colors--canary.295.4071317925.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.295.4071317925.xml)  
[color_metro_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.295.4071317925.swift)  
[color_metro_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.295.4071317925.kt)  
[color_metro_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.295.4071317925.ts)  
[color_sbermarket_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.295.4071317925.swift)  
[color_sbermarket_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.295.4071317925.kt)  
[color_sbermarket_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.295.4071317925.ts)  
[color_sberprime_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.295.4071317925.swift)  
[color_sberprime_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.295.4071317925.kt)  
[color_sberprime_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.295.4071317925.ts)  
[color_selgros_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.295.4071317925.swift)  
[color_selgros_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.295.4071317925.kt)  
[color_selgros_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.295.4071317925.ts)  
[color_smbusiness_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.295.4071317925.swift)  
[color_smbusiness_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.295.4071317925.kt)  
[color_smbusiness_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.295.4071317925.ts)  
[PlasmaTokensColor--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.295.4071317925.swift)  
[shadow_sbermarket_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.295.4071317925.ts)  
[typo_mage_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.295.4071317925.swift)  
[typo_mage_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.295.4071317925.kt)  
[typo_mage_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.295.4071317925.ts)  
[typo_plasma_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.295.4071317925.swift)  
[typo_plasma_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.295.4071317925.kt)  
[typo_plasma_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.295.4071317925.ts)  
[typo_ruler_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.295.4071317925.swift)  
[typo_ruler_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.295.4071317925.kt)  
[typo_ruler_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.295.4071317925.ts)  
[typo_sage_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.295.4071317925.swift)  
[typo_sage_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.295.4071317925.kt)  
[typo_sage_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.295.4071317925.ts)  
[typo_sbermarket_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.295.4071317925.swift)  
[typo_sbermarket_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.295.4071317925.kt)  
[typo_sbermarket_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.295.4071317925.ts)  
[typo_soulmate_ios-swift--canary.295.4071317925.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.295.4071317925.swift)  
[typo_soulmate_kotlin--canary.295.4071317925.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.295.4071317925.kt)  
[typo_soulmate_react-native--canary.295.4071317925.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.295.4071317925.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.133.0-canary.295.4071317925.0
  # or 
  yarn add @salutejs/plasma-b2c@1.133.0-canary.295.4071317925.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-295/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-295/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-295/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-295/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-295/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-295/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-295/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-295/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-295/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-295/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-295/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-295/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-295/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-295/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-295/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-295/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-295/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-295/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-295/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-295/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-295/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-295/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-295/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-295/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-295/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-295/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-295/temple-storybook/

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -12,30 +12,37 @@ import {
...
     border: 0 none;
-    border-radius: 0.75rem;
-    color: ${secondary};
+    border-radius: var(--border-radius);
 
-    padding: 1.25rem 1.5rem;
-    padding-bottom: 3.5rem;
+    font-size: var(--font-size);
+    line-height: var(--line-height);
+    letter-spacing: -0.02em;
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextArea/TextArea.tsx:43]: интересно почему это не в токене

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx [hunk 2]
```diff
@@ -12,30 +12,37 @@ import {
...
     /**
      * Вспомогательные тексты снизу для поля ввода.
      */
     leftHelper?: string;
     rightHelper?: string;
+
+    /**
+     * Возможные размеры компонента
+     */
+    size?: 'l' | 'm' | 's' | 'xs';
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextArea/TextArea.tsx:29]: интересно мы всегда пишем с большего к меньшему?

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -80,6 +115,7 @@ export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
             style,
             id,
             className,
+            size = 'm',
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextArea/TextArea.tsx:118]: я же правильно понимаю что размер m не меняет вид текущего компонента?

Comment (Yeti-or) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:118]: хотя судя по скриншотам `cypress/snapshots/b2c/components/TextArea/TextAreaBase.component-test.tsx/plasma-core TextArea -- success, warning, error.snap.png`
меняет =/

раньше был size=s ? 
или был вообще такой которого нет?

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:118]: @Yeti-or 

> или был вообще такой которого нет?

Предыдущие размеры не попадают под новую размерную сетку. Так что тут ответ - такого размера нет.

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:118]: @Yeti-or К сожалению размеры по высоте меняются. 

Я дал песочницу посмотреть @gibushnev и он говорит что все ок.

Я правильно тебя понимаю, что изменение размеров не очень история?

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx (hunk 2 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:29]: @Yeti-or Я так всегда пишу ) 

По другому нужно? )

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:43]: @Yeti-or Что именно ты имеешь ввиду?

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-295/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-295/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-295/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-295/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-295/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-295/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-295/temple-storybook/

File: packages/plasma-b2c/src/components/TextArea/TextArea.props.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        --top: 1.188rem;
+        --right: 1.125rem;
+
+        --padding-x: 1.125rem;
+        --padding-right: 3.625rem;
+
+        --padding-bottom: 0.75rem;
+        --padding-bottom-override: 2.375rem;
+
+        --padding: 1.063rem var(--padding-x) var(--padding-bottom) var(--padding-x);
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextArea/TextArea.props.ts:14]: не понял чем это отличается от предыдущих padding и зачем его в одно св-во запихивать, если можно точечно раставить padding

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:29]: я посмотрел что у нас то так то так

File: packages/plasma-b2c/src/components/TextArea/TextArea.props.ts (hunk 4 above)
Comment (Yeti-or) [reply packages/plasma-b2c/src/components/TextArea/TextArea.props.ts:14]: хотя я думаю так тоже ок

**Review by Yeti-or — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**